### PR TITLE
changes for receipt endpoint https://github.com/bcgov/entity/issues/653

### PR DESCRIPTION
**Pull request by saravankumarpa**
*Issue #:*

code changes for https://github.com/bcgov/entity/issues/653
*Description of changes:*

Two endpoints for receipt generation added.
Template did some minor changes.
Flake print added

*API Code Review Checklist:*
 - #### Common
    - [x] Proper naming conventions followed on variables, classes, definitions etc.
    - [x] Proper logging levels are used in the code 
 - #### REST 
    - [x] Endpoints follows agreed pattern and organized around resources
    - [x] Conform to http semantics
    - [x] Proper filtering and pagination is provided for large amounts of data
 - #### Code structure
    - [x] Resources is used for request payload processing, validations, response codes, messages etc.
    - [x] Service is used for business logic, 3rd party communication management. Maintains proper folder structure.
    - [x] Model is used for database models.
    - [x] Schemas is used for json schema
    - [x] Util contains commonly used utility functions, constants
 - #### Test Cases
    - [x] Unit test cases with happy and sad path for service, model (?) layers
    - [x] Endpoints test cases with happy and sad path for resources
    - [x] Mocking of services (mainly 3rd party dependencies) provided if applicable
    - [x] Postman collection tests provided for endpoints with happy path
 - #### Code Quality
    - [x] Proper code coverage for all the layers (percentage 80, 90 ?)
    - [x] Linting
- #### DB Migrations
    - [ ] Alembic is used to manage database migration. [Reference](https://github.com/bcgov/namex/blob/master/docs/database.md)

*UI Code Review Checklist:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the namex license (Apache 2.0).

### CODE REVIEW

**Comment by saravankumarpa** [comment]
Tests were working ;but failing now because sumesh is working on PAyBc changes.They should work without changes when he is done /or I will make those changes if necessary.

Flake 8
image

Pylint
image

File: pay-api/src/pay_api/resources/invoice_receipt.py [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+
+from pay_api import jwt as _jwt
+from pay_api.exceptions import BusinessException
+from pay_api.schemas import utils as schema_utils
+from pay_api.services import ReceiptService
+from pay_api.utils.enums import Role
+from pay_api.utils.util import cors_preflight
+
+
+API = Namespace('Invoice receipts', description='Payment System - Receipts')
```
Comment (sutherlanda) [inline pay-api/src/pay_api/resources/invoice_receipt.py]: Can we make this lower case: 'invoice receipts' to match the other namespaces?

File: pay-api/tests/unit/api/base_test.py [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+}
+
+
+def get_claims(role: str = Role.BASIC.value):
+    """Return the claim with the role param."""
+    claim = {
+        'jti': 'a50fafa4-c4d6-4a9b-9e51-1e5e0d102878',
+        'exp': 31531718745,
+        'iat': 1531718745,
+        'iss': 'https://sso-dev.pathfinder.gov.bc.ca/auth/realms/fcf0kpqr',
```
Comment (sutherlanda) [inline pay-api/tests/unit/api/base_test.py]: Will this always come from the dev instance? Should this be pulled from env?

Comment (saravankumarpa) [reply pay-api/tests/unit/api/base_test.py]: I think so..But as an improvement , can be configured in the environment variables/configs as well...
@pksumesh thoughts?

File: pay-api/src/pay_api/resources/payment_receipt.py [hunk 3]
```diff
@@ -0,0 +1,61 @@
+# Copyright © 2019 Province of British Columbia
```
Comment (pksumesh) [inline pay-api/src/pay_api/resources/payment_receipt.py]: Can you check if these can be combined as single resource?

**Comment by saravankumarpa** [comment]
@pksumesh @sutherlanda both changes are done.Plz go ahead with review

**Pull request merged**